ios/scripts/asc_max_build.py

```
@@ -0,0 +1,143 @@
+    ASC `sort=-version` is a STRING sort, so it cannot be trusted across builds
+    with different digit counts (the exact bug this guard exists to prevent).
+    Fetch pages and compute the max as an integer instead.
+    """
+    highest = 0
+    saw_any = False
+    path = f"/v1/builds?filter[app]={app_id}&fields[builds]=version&limit=200"
+    pages = 0
+    while path and pages < 20:
+        status, body = _api(token, path)
+        if status != 200:
+            raise RuntimeError(f"builds lookup HTTP {status}: {json.dumps(body)[:300]}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)